lib/story_query.rb

```diff
@@ -1,34 +1,30 @@
 require_dependency 'user_query'
 
 class StoryQuery
-
   # Returns a list of stories having the given IDs.
   def self.find_by_ids(story_ids, current_user)
-    stories = Story.where(id: story_ids).for_user(current_user)
-                   .includes(:user, :substories, :target,
-                             substories: [:user, :target])
+    stories = Story.where(id: story_ids)
+              .for_user(current_user)
+              .includes(:user, :substories, :target,
+                        substories: %i(user target))
 
     # Preload genres for anime stories.
-    anime_stories = stories.select {|s| s.target_type == "Anime" }.map(&:target)
-                   .index_by(&:id)
-    Anime.where(id: anime_stories.values.map(&:id)).includes(:genres).each do |a|
-      assoc = anime_stories[a.id].association(:genres)
-      assoc.loaded!
-      assoc.target.concat(a.genres)
-      a.genres.each {|genre| assoc.set_inverse_instance(genre) }
-    end
+    stories.select { |s| s.target_type == 'Anime' }.map(&:target).index_by(&:id)
```

Comment:
We shouldn't remove this unless we know for certain that it won't introduce an N+1 query.  I'm not sure the eager loading in the default scope will save us here with nested associations (stories -> anime -> genres) especially when they're both polymorphic...